go/src/url_shortener/stats.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// StatsJSON Statistic data ready for JSON serialization
+type StatsJSON struct {
+	ServerStats serverStatsJSON `json:"server_stats"`
+
+	mux sync.Mutex
```

Comment:
Here a mutex works, but each statistics could also be updated atomically, which would be better performance wise in case our server was under heavy load, when every microsecond counts.

Have a look at the [atomic package](https://godoc.org/sync/atomic)